flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/BoundedStreamTask.java

```diff
@@ -72,6 +74,10 @@ protected void init() throws Exception {
 		// re-initialize the operator with the correct collector.
 		StreamOperatorFactory<OUT> operatorFactory = configuration.getStreamOperatorFactory(getUserCodeClassLoader());
 		headOperator = operatorFactory.createStreamOperator(this, configuration, new CollectorWrapper<>(collector));
+		if (operatorFactory instanceof YieldingOperatorFactory) {
```

Comment:
Should this block go before the previous line (`operatorFactory.createStreamOperator(this, configuration, new CollectorWrapper<>(collector));`)?